tiledb/sm/filesystem/ls_callback.h

```diff
@@ -0,0 +1,87 @@
+/**
+ * @file ls_callback.h
```

Comment:
This file needs a new name, and some of its contents do also.

`callback` is a word specific to C. We don't want to be using it for C++ objects. We will need a constructor argument called `callback` to handle the C API (and a related member variable), but that's about the extent to which we want to use this word.

With C++ `<algorithm>` they're not called "callbacks" because they usually do _not_ call back into the caller, even as they call into other code provided by the call. Look, say, at https://en.cppreference.com/w/cpp/algorithm/copy in the signature of `copy_if`, where the template argument is `UnaryPredicate`. The template arguments we use should be similarly named.

Reply:
Renaming the file is fine, but
> callback is a word specific to C

"callback" is commonly used in Javascript and many other languages; it came up multiple times during a JS UI discussion just yesterday.

https://en.wikipedia.org/wiki/Callback_(computer_programming)

https://en.wikipedia.org/wiki/Callback_(computer_programming)